app/controllers/view_data/departments_controller.rb

```diff
@@ -1,7 +1,15 @@
 module ViewData
   class DepartmentsController < ViewDataController
+    def extract_natural_key
```

Comment:
I think this can safely (assuming :id is in the hash) just do ```params[:id].split('#').first``` which will work whether id has a fragment or not.

Reply:
Might want to move it to ViewDataController too if it's used by more than one controller.